src/migrate.ts

```diff
@@ -53,21 +53,24 @@ export async function migrate(
 ) {
   const { escapedWorkerSchema } = processSharedOptions(options);
   let latestMigration: number | null = null;
-  try {
-    const {
-      rows: [row],
-    } = await client.query(
-      `select id from ${escapedWorkerSchema}.migrations order by id desc limit 1;`,
-    );
-    if (row) {
-      latestMigration = row.id;
-    }
-  } catch (e) {
-    if (e.code === "42P01") {
-      await installSchema(options, client);
-    } else {
-      throw e;
-    }
+
+  const {
+    rows: [schemaExists],
+  } = await client.query({
+    text: `SELECT EXISTS(SELECT 1 FROM information_schema.schemata WHERE schema_name=$1);`,
+    values: [escapedWorkerSchema],
+  });
+  if (!schemaExists.exists) {
+    await installSchema(options, client);
+  }
```

Comment:
It's more performant to query pg_namespace than information_schema; I've also edited the code to be more consistent with the rest of the codebase. You shouldn't use `escapedWorkerSchema` in a value position because it's escaped as an identifier (so for `MySchema` it'd have the value `"MySchema"` (with quotes)), instead use the `workerSchema` value directly.

```suggestion
  const {
    rows: [pgNamespace],
  } = await client.query(
    `select oid from pg_catalog.pg_namespace where nspname = $1;`,
    [workerSchema],
  );
  if (!pgNamespace) {
    await installSchema(options, client);
  }
```